lib/class-wp-rest-global-styles-controller-gutenberg.php

```diff
@@ -458,14 +497,219 @@ protected function get_available_actions( $post, $request ) {  // phpcs:ignore V
 	}
 
 	/**
-	 * Retrieves the query params for the global styles collection.
+	 * Checks if a given request has access to read global styles.
 	 *
-	 * @since 5.9.0
+	 * @since 7.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) {
+		if ( current_user_can( 'edit_theme_options' ) ) {
+			return true;
+		}
+
+		return new WP_Error(
+			'rest_cannot_read_global_styles',
+			__( 'Sorry, you are not allowed to access global styles.', 'gutenberg' ),
+			array( 'status' => rest_authorization_required_code() )
+		);
+	}
+
+	/**
+	 * Retrieves a collection of style variations.
+	 *
+	 * Returns all wp_global_styles posts that are style variations
+	 * (identified by _wp_style_variation meta).
```

Comment:
Is this the same value as the PHP const below? Does that mean it should be configurable?

I'm looking at:

`define( 'GUTENBERG_STYLE_VARIATION_META_KEY', '_wp_style_variation' );`